### DISCUSSION TITLE
STM32F401 CDC_ACM Issues

### DISCUSSION THREAD

**Discussion by MWP** [hathach/tinyusb#1156] [Q&A]
Greetings all,
I'm attempting to get tinyusb working with my STM32F401 based project.
Device only, one interface (CDC ACM). I'm using unmodified dcd_synopsys.c .

The virtual COM port is setup by the Win10 driver correctly, and EP0 works as it should (as far as i can tell).
On opening the COM port in a simple terminal app the debug output shows:
```
00> USBD Setup Received 21 22 03 00 00 00 00 00 
00>   CDC control request
00>   Queue EP 80 with 0 bytes ...
00> USBD Xfer Complete on EP 80 with 0 bytes
00>   Set Control Line State: DTR = 1, RTS = 1
```

The problem comes when trying to send data to my device. OTG_FS_IRQHandler() is not fired.
I've checked GINTSTS after attempting to write data to my device. USB_OTG_GINTSTS_RXFLVL is never set.

Does anyone know what the problem may be here?

Thanks in advance!

**Comment by hathach** [comment]
have you tried the stock example `cdc_msc`. If not please try with it first to see if that works. also if you have one of the supported boards, you should run on those as well just rule out any possible hardware issue.
